python/arcticdb/adapters/s3_library_adapter.py

```diff
@@ -127,10 +141,13 @@ def config_library(self):
             ca_cert_path=self._ca_cert_path,
             ca_cert_dir=self._ca_cert_dir,
             ssl=self._ssl,
+            aws_auth=self._query_params.aws_auth,
+            aws_profile=self._query_params.aws_profile,
+            native_cfg=self.native_cfg,
         )
 
         lib = NativeVersionStore.create_store_from_config(
-            env_cfg, _DEFAULT_ENV, CONFIG_LIBRARY_NAME, encoding_version=self._encoding_version
+            env_cfg, _DEFAULT_ENV, CONFIG_LIBRARY_NAME, encoding_version=self._encoding_version, native_cfg=self.native_cfg
```

Comment:
Similar here, can we just change the type of `env_cfg` rather than adding the new `native_cfg` parameter? (in a backwards compatible way)